docs/reference/sql-openapi/api.md

```diff
@@ -402,4 +402,59 @@ Limit's values can be customized through configuration:
 By default *offset* is not applied to the request.
 Clients can override this behavior by passing a value.
 
-*Offset* only accepts values `>= 0`. Otherwise an error is return.
\ No newline at end of file
+*Offset* only accepts values `>= 0`. Otherwise an error is return.
+
+## Allow the primary keys in the input
+
+`@platformatic/sql-openapi` allows for specifying if to accept the table primary keys
+in the inputs to the various routes.
+
+To configure:
+
+```javascript
+app.register(require('@platformatic/sql-openapi'), {
+  allowIdInInput: false
+})
+```
+
+_Example_
+
+If `allowIdInInput` is set to `false`:
+
+```
+$ curl -X 'POST' \
+  'http://localhost:3042/pages/' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "id": 42,
+  "title": "Hello Platformatic!",
+  "body": "Welcome to Platformatic!"
+}'
+
+{
+  "id": 1,
```

Comment:
IMO, if `allowIdInInput` is set to `false` it should generate a route schema without an `id` prop and here it should throw a 400 because data has an ddditional property. And if you do it programatically (with sql-mapper) it also should throw an error. Otherwise it looks like a place for a very nasty bug. wdyt?

Reply:
That’s how the schemas are generated. However Ajv/Fastify do not error in this case, but they rather filter the additional property.

Reply:
1. Then IMHO it should throw an error on a `sql-mapper` level.
2. It's easy to create a schema that would fail if some known property exist. Just set `false` for a property schema.

Reply:
My problem is that it would be really hard to notice the problem if you passing an id and `allowPrimaryKeyInInput` set to false. If you think that skipping is the best behavior it should at least log a warn message.

Reply:
It’s not possible to do it in sql-mapper because I’ve implemented .save() as an upsert.

Reply:
> It's easy to create a schema that would fail if some known property exist. Just set false for a property schema.

Oh I’ll try this!

Reply:
> Oh I’ll try this!

Or you can do this. Choose the error message you like more.

```json
{
  "type": "object",
  "not": {
    "required": ["foo"]
  }
}
```

Reply:
Done with a slightly different syntax that gave a better error message.